### ISSUE TITLE
shellshock: crux

### ISSUE THREAD

**Issue by tianon** [docker-library/official-images#479]
```console
$ ./bashbrew/bashbrew.sh list crux | xargs ./test/run.sh -t shellshock
testing crux:latest
	'shellshock' [1/1]...shellshock_test.sh: line 17:    13 Segmentation fault      shellshocker="() { x() { _;}; x() { _;} <<a; }" bash -c date 2> /dev/null
shellshock_test.sh: line 50:    30 Segmentation fault      bash -c 'true <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF' 2> /dev/null
failed
testing crux:3.1
	'shellshock' [1/1]...shellshock_test.sh: line 17:    13 Segmentation fault      shellshocker="() { x() { _;}; x() { _;} <<a; }" bash -c date 2> /dev/null
shellshock_test.sh: line 50:    30 Segmentation fault      bash -c 'true <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF' 2> /dev/null
failed
```

**Comment by tianon** [comment]
ping @therealprologic

**Comment by therealprologic** [comment]
@tianon hmm?

**Comment by tianon** [comment]
If you run https://github.com/wreiske/shellshocker/blob/master/shellshock_test.sh inside the current CRUX image, you'll see that it's vulnerable:
```console
$ docker pull crux
511136ea3c5a: Already exists 
e1c4fee8ca32: Already exists 
7a73a3cc03b3: Already exists 
crux:latest: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.
Status: Image is up to date for crux:latest
$ docker run -it --rm crux
# curl -sSL 'https://raw.githubusercontent.com/wreiske/shellshocker/master/shellshock_test.sh' | bash
CVE-2014-6271 (original shellshock): VULNERABLE
bash: line 17:    16 Segmentation fault      shellshocker="() { x() { _;}; x() { _;} <<a; }" bash -c date 2> /dev/null
CVE-2014-6277 (segfault): VULNERABLE
CVE-2014-6278 (Florian's patch): VULNERABLE
CVE-2014-7169 (taviso bug): VULNERABLE
bash: line 50:    33 Segmentation fault      bash -c 'true <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF' 2> /dev/null
CVE-2014-7186 (redir_stack bug): VULNERABLE
CVE-2014-7187 (nested loops off by one): not vulnerable
CVE-2014-//// (exploit 3 on http://shellshocker.net/): not vulnerable
```

**Comment by therealprologic** [comment]
Seems I've forgotten to rebuild the image :/

```
prologic@daisy
Wed Feb 11 06:08:37 
~
$ ./test-shellshock.sh 
CVE-2014-6271 (original shellshock): not vulnerable
CVE-2014-6277 (segfault): not vulnerable
CVE-2014-6278 (Florian's patch): not vulnerable
CVE-2014-7169 (taviso bug): not vulnerable
CVE-2014-7186 (redir_stack bug): not vulnerable
CVE-2014-7187 (nested loops off by one): not vulnerable
CVE-2014-//// (exploit 3 on http://shellshocker.net/): not vulnerable
```

We updated Bash a while ago in our ports. I'll update the image straight away :)

**Comment by tianon** [comment]
Thanks! :heart:

**Comment by therealprologic** [comment]
PR #492 should fix this :)

```
#!bash
prologic@daisy
Sun Feb 15 19:57:56 
~
$ dki crux:3.1
bash-4.3# cat - > test-shellshock.sh
#!/bin/bash
EXITCODE=0

# CVE-2014-6271
CVE20146271=$(env 'x=() { :;}; echo vulnerable' 'BASH_FUNC_x()=() { :;}; echo vulnerable' bash -c "echo test" 2>&1 | grep 'vulnerable' | wc -l)

echo -n "CVE-2014-6271 (original shellshock): "
if [ $CVE20146271 -gt 0 ]; then
	echo -e "\033[91mVULNERABLE\033[39m"
	EXITCODE=$((EXITCODE+1))
else
	echo -e "\033[92mnot vulnerable\033[39m"
fi

# CVE-2014-6277
# it is fully mitigated by the environment function prefix passing avoidance
CVE20146277=$((shellshocker="() { x() { _;}; x() { _;} <<a; }" bash -c date 2>/dev/null || echo vulnerable) | grep 'vulnerable' | wc -l)

echo -n "CVE-2014-6277 (segfault): "
if [ $CVE20146277 -gt 0 ]; then
	echo -e "\033[91mVULNERABLE\033[39m"
	EXITCODE=$((EXITCODE+2))
else
	echo -e "\033[92mnot vulnerable\033[39m"
fi

# CVE-2014-6278
CVE20146278=$(shellshocker='() { echo vulnerable; }' bash -c shellshocker 2>/dev/null | grep 'vulnerable' | wc -l)

echo -n "CVE-2014-6278 (Florian's patch): "
if [ $CVE20146278 -gt 0 ]; then
	echo -e "\033[91mVULNERABLE\033[39m"
	EXITCODE=$((EXITCODE+4))
else
	echo -e "\033[92mnot vulnerable\033[39m"
fi

# CVE-2014-7169
CVE20147169=$((cd /tmp; rm -f /tmp/echo; env X='() { (a)=>\' bash -c "echo echo nonvuln" 2>/dev/null; [[ "$(cat echo 2> /dev/null)" == "nonvuln" ]] && echo "vulnerable" 2> /dev/null) | grep 'vulnerable' | wc -l)

echo -n "CVE-2014-7169 (taviso bug): "
if [ $CVE20147169 -gt 0 ]; then
	echo -e "\033[91mVULNERABLE\033[39m"
	EXITCODE=$((EXITCODE+8))
else
	echo -e "\033[92mnot vulnerable\033[39m"
fi

# CVE-2014-7186
CVE20147186=$((bash -c 'true <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF <<EOF' 2>/dev/null || echo "vulnerable") | grep 'vulnerable' | wc -l)

echo -n "CVE-2014-7186 (redir_stack bug): "
if [ $CVE20147186 -gt 0 ]; then
	echo -e "\033[91mVULNERABLE\033[39m"
	EXITCODE=$((EXITCODE+16))
else
	echo -e "\033[92mnot vulnerable\033[39m"
fi

# CVE-2014-7187
CVE20147187=$(((for x in {1..200}; do echo "for x$x in ; do :"; done; for x in {1..200}; do echo done; done) | bash || echo "vulnerable") | grep 'vulnerable' | wc -l)

echo -n "CVE-2014-7187 (nested loops off by one): "
if [ $CVE20147187 -gt 0 ]; then
	echo -e "\033[91mVULNERABLE\033[39m"
	EXITCODE=$((EXITCODE+32))
else
	echo -e "\033[92mnot vulnerable\033[39m"
fi

# CVE-2014-////
CVE2014=$(env X=' () { }; echo vulnerable' bash -c 'date' | grep 'vulnerable' | wc -l)

echo -n "CVE-2014-//// (exploit 3 on http://shellshocker.net/): "
if [ $CVE2014 -gt 0 ]; then
	echo -e "\033[91mVULNERABLE\033[39m"
	EXITCODE=$((EXITCODE+64))
else
	echo -e "\033[92mnot vulnerable\033[39m"
fi

exit $EXITCODE
bash-4.3# chmod +x test-shellshock.sh 
bash-4.3# ./test-shellshock.sh 
CVE-2014-6271 (original shellshock): not vulnerable
CVE-2014-6277 (segfault): not vulnerable
CVE-2014-6278 (Florian's patch): not vulnerable
CVE-2014-7169 (taviso bug): not vulnerable
CVE-2014-7186 (redir_stack bug): not vulnerable
CVE-2014-7187 (nested loops off by one): not vulnerable
CVE-2014-//// (exploit 3 on http://shellshocker.net/): not vulnerable
bash-4.3# 
```

**Comment by yosifkit** [comment]
This was fixed in #492.
